### PR TITLE
Incorporated index "clearing" functionality and added to CacheAllPagesJob

### PR DESCRIPTION
**Pull request by mdjward**

### CODE REVIEW

File: lib/ElasticSearch/Event/EventManager.php [hunk 1]
```diff
@@ -0,0 +1,134 @@
...
+
+                    $target->registerJob(
+                        new Schedule_Maintenance_Job(
+                            self::MAINTENANCE_JOB_REBUILD_PAGES,
+                            $this->cacheAllPagesJob,
+                            'rebuildPageCache'
+                        )
+                    );
+                },
+                $this
```
Comment (seeruk) [inline lib/ElasticSearch/Event/EventManager.php:80]: Would `function(Event $event) use ($this as $foo) {` work?

File: lib/ElasticSearch/Processor/Page/ElementProcessor.php [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+     * 
+     * @param Document_Tag $tag
+     * @return string
+     * @throws ProcessorException
+     */
+    public function processElement(Document_Tag $tag)
+    {
+        $elementData = $tag->getData();
+        
+        if (!is_string($elementData) || ($elementData = trim($elementData)) === '') {
```
Comment (seeruk) [inline lib/ElasticSearch/Processor/Page/ElementProcessor.php:55]: Maybe move the trim up to line 53, so you don't have to assign in the if?

File: lib/ElasticSearch/Event/EventManager.php (hunk 1 above)
Comment (mdjward) [inline lib/ElasticSearch/Event/EventManager.php:80]: Sadly not; I have tried this before.  The only other option I'm aware of is to initialise a variable which isn't a reserved keyword like $this (say, $self) and assign it the $this reference, then "use" $self.  Sucks!

Whereas you could just bind the closure; really useful PHP5.4+ feature.

File: lib/ElasticSearch/Processor/Page/ElementProcessor.php (hunk 2 above)
Comment (mdjward) [inline lib/ElasticSearch/Processor/Page/ElementProcessor.php:55]: By that point, we haven't established whether or not the mixed element data is returned as a string or a nested object.  This is why the is_string check occurs first; then an assignment-in-expression is used to trim what we now know to be string data for subsequent statements.

**Pull request merged**